### PR TITLE
Added GPU Temperature and utilization

### PR DESCRIPTION
**Pull request by JeremiahPetersen**
This pull request introduces the following changes:

1. AMD Support: Implemented a check for the availability of the pyadl library and the accessibility of AMD drivers. This ensures that the application only attempts to retrieve AMD GPU details if they are accessible.
2. Temperature Information: A new section, "Temperature," has been added to the GUI to display the temperature details of the CPU and GPUs (both NVIDIA and AMD). It uses psutil, GPUtil, pyadl, and wmi to fetch the temperature details based on the system and platform.
3. Error Handling in Graphics Section: Enhanced the error handling in the graphics section to display a message if there's an issue fetching the AMD GPU details.

Please review and let me know of any feedback or changes required.

### CODE REVIEW

**Comment by diamant3** [comment]
Thank you so much for your PR!

By the way, I based the SYSIG app into a major parts of a computer, please don't separate the section for minor section like temperature, I suggest to add that temp info to processor section and Graphics section and please minimize the use of modules as possible and lastly, run `pip install -r requirements.txt` first for every OS. 

Sorry for my long and demanding request because I don't have any build workflow(just a linter 😢) for every possible OS to test.

**Comment by JeremiahPetersen** [comment]
no worries at all, thank you for the feedback!  I will work on these adjustments in the next 24 hours :)

**Comment by diamant3** [comment]
Okay I will review that asap, thanks again!

**Comment by JeremiahPetersen** [comment]
I added GPU temp (updates every second) as a bullet in Graphics
I added CPU temp (updates every second) as a bullet in Processor
GPU temp is working on all tests so far
CPU temp is not updating every second as expected on windows platforms. (mechanism is working, so I am working to find another solution to this today.)

**Comment by diamant3** [comment]
Ok, let me know if it's ready for review. Thanks!

**Comment by JeremiahPetersen** [comment]
It is ready for review now.  The gpu temp and utilization are working.  The cpu temp is working, but for some windows / intel systems, it would need additional software (Open Hardware Monitor OHM) which would be adding too much and may be unnecessary.  I cleaned up the code.  Let me know what changes you would like.

File: requirements.txt [hunk 1]
```diff
@@ -1,6 +1,7 @@
+psutil
+humanize
 dearpygui
-py-cpuinfo
+cpuinfo
```
Comment (diamant3) [inline requirements.txt]: This is py-cpuinfo not cpuinfo

File: sysig.py [hunk 2]
```diff
@@ -110,19 +173,63 @@ def get_cpu_util():
...
+                except (ImportError, Exception) as e:
+                    dpg.add_text(f"Error fetching NVIDIA GPU temperature: {e}", bullet=True, parent=gpu_temp_placeholder)
+
+                # AMD GPU temperatures
+                if AMD_SUPPORTED:
+                    try:
+                        amd_manager = pyadl.ADLManager.getInstance()
+                        devices = amd_manager.getDevices()
+                        for device in devices:
+                            temperature_data = device.getTemperature()
```
Comment (diamant3) [inline sysig.py]: I think it's `getCurrentTemperature()`, please check their docs if correct

File: sysig.py [hunk 3]
```diff
@@ -33,7 +50,52 @@ def get_cpu_util():
...
+            pass
+
+        try:
+            # Fallback to WMI approach if psutil does not provide the info
+            import wmi
+            w = wmi.WMI(namespace="root\\wmi")
+            temperature_info = w.MSAcpi_ThermalZoneTemperature()[0]
+            return temperature_info.CurrentTemperature / 10.0 - 273.15
+        except wmi.x_access_denied:
+            return "Access Denied (Run script as administrator)"
```
Comment (diamant3) [inline sysig.py]: Is there any solution for this without using administrator access in Windows? All of the modules I used don't need administrator access.

**Review by diamant3 — CHANGES_REQUESTED** [review]
I've reviewed it all now. I will run the workflow and merge it later if it's running correctly.

File: sysig.py (hunk 2 above)
Comment (JeremiahPetersen) [reply sysig.py]: I checked here - https://github.com/nicolargo/pyadl - and you are correct.  I made the changes.

File: sysig.py (hunk 3 above)
Comment (JeremiahPetersen) [reply sysig.py]: That was the problem I was running into, I researched the problem for a while and the solutions were to run wmi using admin access, or use additional software like OHM.  As the graphics additions are working, I can scrap the cpu temp idea until I can figure out a better solution for Windows / Intel.  I can remove the cpu temperature code for now if you would like to incorporate just the gpu additions.

Comment (diamant3) [reply sysig.py]: > use additional software like OHM.

Please don't, This is just a simple app 😅

> I can remove the cpu temperature code for now if you would like to incorporate just the gpu additions.

If it's okay with you, Please do. It's better to do GPU additions for now, and you can submit another PR if you have a solution for admin access.

Comment (JeremiahPetersen) [reply sysig.py]: lol yea I like that it is a simple app.  I was trying to implement an easy way to see cpu temp but it turned out to be a rabbit hole lol.  I like keeping it simple.  That sounds good.  I will edit the code and send that in a moment

Comment (diamant3) [reply sysig.py]: Thanks! I will run the workflow and merge this later.

**Comment by JeremiahPetersen** [comment]
This is good for review, I removed the cpu temp and will continue researching to see if it is possible to do in a simple way.  If not, no worries.  The gpu additions should all be working without issue now.  Let me know if there are any other changes to make.  Thanks for letting me contribute and providing clear direct feedback!

File: sysig.py (hunk 2 above)
Comment (diamant3) [reply sysig.py]: Okay, thanks!

**Comment by diamant3** [comment]
Don't worry it's just errors for variable naming, length of line etc., I'll merge this later. thanks again!

**Comment by JeremiahPetersen** [comment]
no worries!  I was going through each of the errors and realizing that lol

**Comment by diamant3** [comment]
Thanks!

**Comment by JeremiahPetersen** [comment]
You are welcome!  Thank you!

**Pull request merged**